### ISSUE TITLE
Get Rid of `GcovTestObserver`

### ISSUE THREAD

**Issue by fabb** [SnapKit/Masonry#394]
`XCTestObserver` is deprecated, so another way should be found for unit tests. As far as I have understood this is just needed for generating unit test coverage reports, is that correct? In that case maybe the newer ways of Xcode can be used instead to generate that reports: https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/testing_with_xcode/chapters/07-code_coverage.html

**Comment by robertjpayne** [comment]
@fabb is this still relevant after #392 merging?

**Comment by fabb** [comment]
Yes, actually #392 was the motivation for this issue. See line 500 in the travis build log: https://travis-ci.org/SnapKit/Masonry/builds/168862088#L500

```
⚠️  /Users/travis/build/SnapKit/Masonry/Tests/GcovTestObserver.m:11:31: 'XCTestObserver' is deprecated [-Wdeprecated-declarations]
```
